### PR TITLE
Fixed number/integer generators (via random)

### PR DESCRIPTION
**Pull request by pateketrueke**

### CODE REVIEW

**Comment by JonathanMontane** [comment]
This fix will not cover this case:

```
{
  "type": "number",
  "minimum": 1,
  "maximum": 1.05,
  "maximumExclusive": true
}
```

nor 

```
{
  "type": "number",
  "minimum": 1,
  "maximum": 2,
  "minimumExclusive": true,
  "maximumExclusive": true
}

**Comment by pateketrueke** [comment]
Oh, that edge cases... let test your solution too, thanks for your feedback!

File: lib/types/number.js [hunk 1]
```diff
@@ -1,16 +1,20 @@
...
+    // otherwise having values like { min: 1, max: 1.5 } will produce invalid values (NaN)
+    var fixedUnit = ((min % 1) === 0 && (max % 1) === 0) ? 1 : FLOAT_DIFFERENCE;
     if (value.exclusiveMinimum && value.minimum && min === value.minimum) {
-        min += multipleOf || 1;
+        min += multipleOf || fixedUnit;
     }
     if (value.exclusiveMaximum && value.maximum && max === value.maximum) {
-        max -= multipleOf || 1;
+        max -= multipleOf || fixedUnit;
     }
```
Comment (JonathanMontane) [inline lib/types/number.js]:
```
{
  "type": "number",
  "minimum": 1,
  "maximum": 2,
  "maximumExclusive": true,
  "minimumExclusive": true
}
```
will give us the following values for `fixedUnit`, `min` and `max`:
```
fixedUnit = 1,
min = 2,
max = 1
```
which will return `NaN`, despite being a well-defined interval.

Changing the step is not the solution to this issue.

Comment (pateketrueke) [inline lib/types/number.js]: Which value do you expect to being generated giving those constraints?

Comment (JonathanMontane) [inline lib/types/number.js]:
```
1.00102
1.20093
1.60043
...
```

Any decimal value between `1` and `2`, but neither `1` nor `2`

Comment (pateketrueke) [inline lib/types/number.js]: Decimals are perfect for `number` values which stand for non integers, but applying the same rules exactly but for `integer` instead which value would output?

Comment (JonathanMontane) [inline lib/types/number.js]: oh, if your schema is the following:

```
{
  "type": "integer",
  "minimum": 1,
  "maximum": 2,
  "maximumExclusive": true,
  "minimumExclusive": true
}
```

then the result should be `NaN`, as there are no valid integer in that interval

Comment (pateketrueke) [inline lib/types/number.js]: That was my best guess, I'll improve the solution for numbers whilst `NaN` is OK I think we could throw an exception for that, let me see what can I do, thanks for your advice! :beer:

**Comment by ducin** [comment]
what is the status of this?

**Comment by ducin** [comment]
@pateketrueke, @JonathanMontane I have merged this PR with current master. Please finish this PR soon, we're drowning with issues and PRs... ;)

**Comment by pateketrueke** [comment]
Update: cannot merge due outdated commits.
